src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -3157,3 +3166,86 @@ class DatePartFunctionNameSegment(BaseSegment):
 
     type = "function_name"
     match_grammar = Sequence("DATEADD")
+
+
+@ansi_dialect.segment()
+class CreateTriggerStatementSegment(BaseSegment):
+    """Create Trigger Statement.
+
+    Taken from specification in https://www.postgresql.org/docs/14/sql-createtrigger.html
+    Edited as per notes in above - what doesn't match ANSI
+    """
+
+    type = "create_trigger"
+
+    match_grammar = Sequence("CREATE", "TRIGGER", Anything())
+
+    parse_grammar = Sequence(
+        "CREATE",
+        "TRIGGER",
+        Ref("TriggerReferenceSegment"),
+        OneOf("BEFORE", "AFTER", Sequence("INSTEAD", "OF")),
+        Delimited(
+            "INSERT",
+            "DELETE",
+            Sequence(
+                "UPDATE",
+                "OF",
+                Delimited(
+                    Ref("ColumnReferenceSegment"),
+                    terminator=OneOf("INSERT", "DELETE", "UPDATE", "ON"),
+                ),
+            ),
+            delimiter="OR",
+            terminator="ON",
+        ),
+        "ON",
+        Ref("TableReferenceSegment"),
+        AnyNumberOf(
+            Sequence(
+                "REFERENCING",
+                "OLD",
+                "ROW",
+                "AS",
+                Ref("ParameterNameSegment"),
+                "NEW",
+                "ROW",
+                "AS",
+                Ref("ParameterNameSegment"),
```

Comment:
The Postgres page specifically gives examples of:
- Only setting the `NEW` reference
- Setting both but the other way around.

Are these definitely not supported in ANSI?

Given this is in an `AnyNumberOf`, couldn't we just do this?:

```suggestion
                "REFERENCING",
                OneOf("OLD", "NEW"),
                "ROW",
                "AS",
                Ref("ParameterNameSegment"),
```

Reply:
This specific part of syntax is not supported by Postgres - I think you're thinking of the Table Referencing.